### PR TITLE
Slam 87 - IMU compass for Orientation

### LINKED ISSUE #87: Use Phidgets IMU compass to obtain orientation
**Description:**
- `phidgets_imu` node does not publish orientation, but we can fuse it from the magnetometer using `imu_magwick_filter`

**Goals:**
- [x] Obtain orientation from phidgets IMU ros topic

**Acceptance Criteria:**
- [x] phidgets IMU ros topic publishes orientation
- [x] Orientation is consistent with earth reference frame
- [x] Imu topic is updated in all our packages appropriately

**Notes:**
- If this works, then I purchased a second IMU for no reason...

### PR DESCRIPTION
**Pull request by erickmu1**
- Added `imu_filter_madgwick` as a node in `phidgets_imu` package (even though `imu_filter_madgwick` is not part of `phidgets`) since it [phidgets launch files] calls it.
- now imu data is published on `imu/data` and no longer on `imu/data_raw` (made changes to reflect this)
- Added config file `imu_mag_toronto.yaml` which represents magnetic properties of Toronto.
  - **NOTE:** we need to add magnetic declination of -10 degrees in `robot_localization`'s `navsat_transform_node` (concerns issue #61)
- Updated `robot_localization`'s `ekf_localization_node` to reflect the availability of absolute orientation

### CODE REVIEW

File: slam-ws/src/phidgets_drivers/imu_filter_madgwick/Makefile [hunk 1]
```diff
@@ -0,0 +1 @@
+include $(shell rospack find mk)/cmake.mk
```
Comment (tcheung99) [inline slam-ws/src/phidgets_drivers/imu_filter_madgwick/Makefile]: Warning on the side that says no newline at the end of file (good practice) 
https://stackoverflow.com/questions/8172246/what-is-the-meaning-of-warning-no-new-line-at-end-of-file

**Review by tcheung99 — APPROVED** [review]

File: slam-ws/src/phidgets_drivers/imu_filter_madgwick/Makefile (hunk 1 above)
Comment (erickmu1) [reply slam-ws/src/phidgets_drivers/imu_filter_madgwick/Makefile]: I agree adding newlines are important (and did update this) but in general we don't have to look into this in 3rd party packages (ie. `imu_filter_madgwick` was not written by us)

File: slam-ws/src/phidgets_drivers/imu_filter_madgwick/cfg/ImuFilterMadgwick.cfg [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+PACKAGE='imu_filter_madgwick'
+
+from dynamic_reconfigure.parameter_generator_catkin import *
+
+gen = ParameterGenerator()
+                                                                    
+gen.add("gain", double_t, 0, "Gain of the filter. Higher values lead to faster convergence but more noise. Lower values lead to slower convergence but smoother signal.", 0.1, 0.0, 1.0) 
+gen.add("zeta", double_t, 0, "Gyro drift gain (approx. rad/s).", 0, -1.0, 1.0) 
+gen.add("mag_bias_x", double_t, 0, "Magnetometer bias (hard iron correction), x component.", 0, -10.0, 10.0)
```
Comment (1999michael) [inline slam-ws/src/phidgets_drivers/imu_filter_madgwick/cfg/ImuFilterMadgwick.cfg:12]: add the -10 degree decline here?

Comment (erickmu1) [reply slam-ws/src/phidgets_drivers/imu_filter_madgwick/cfg/ImuFilterMadgwick.cfg:12]: No, the -10 degree decline is called magnetic declination and not magnetic bias, I already accounted for the bias in the .yaml file

Comment (erickmu1) [reply slam-ws/src/phidgets_drivers/imu_filter_madgwick/cfg/ImuFilterMadgwick.cfg:12]: Also, magnetic declination is added in `navsat_transform_node`, which is not in `dev` yet

Comment (1999michael) [reply slam-ws/src/phidgets_drivers/imu_filter_madgwick/cfg/ImuFilterMadgwick.cfg:12]: ah okok then looks good to me

**Review by 1999michael — APPROVED** [review]
looks good to me

**Pull request merged**